httpserver/server.go

```diff
@@ -58,15 +70,45 @@ func (server *Server) Run() error {
 	if err != nil {
 		return err
 	}
+
 	lsnr, err := net.ListenTCP("tcp", tcpAddr)
 	if err != nil {
 		return err
 	}
-	svr := http.Server{
+
+	svr := &http.Server{
 		Addr:    server.Address,
 		Handler: server.Router,
 	}
-	return svr.Serve(lsnr)
+
+	if server.CertDirectory != "" {
+		certFile := filepath.Join(server.CertDirectory, certName)
+		keyFile := filepath.Join(server.CertDirectory, keyName)
+
+		logrus.Info(fmt.Sprintf("%s: [%s:%s] [%s:%s]", "starting server using TLS", "certFile", certFile, "keyFile", keyFile))
+
+		if server.CaCertFile != "" {
```

Comment:
Hi @mluker , i m seeing @etrexel 's comment here about mounting cert instead of loading inside the application. Wonder if that comment also applicable here?
https://github.com/deislabs/ratify/pull/361#issuecomment-1290964221

Reply:
I think the scenario is a little different with this CA. The CA cert here is from a volume which is mounted from Gatekeepers auto created CA secret. @etrexel  do you have any feedback on this?

Reply:
fyi - mTLS requires deploying in gatekeeper's namespace but it might not always be `gatekeeper-system`